src/modalities/activation_checkpointing.py

```diff
@@ -0,0 +1,24 @@
+from functools import partial
+
+import torch
+from torch.distributed.algorithms._checkpoint.checkpoint_wrapper import (
+    CheckpointImpl,
+    apply_activation_checkpointing,
+    checkpoint_wrapper,
+)
+from torch.distributed.fsdp.fully_sharded_data_parallel import FullyShardedDataParallel as FSDP
+
+from modalities.models.gpt2.gpt2_model import Block
+
+
+def is_module_to_apply_activation_checkpointing(submodule: torch.nn.Module):
+    return isinstance(submodule, Block)
```

Comment:
`Block` is part of the GPT2 Model and hardcoded here. I think it would be better to have this configurable via the config, in case we have different blocks for other models in the future.  

We could think of having a `Blocks` enum in the config and parameterise this via config yaml.